### PR TITLE
doc/nrf/ug_nrf5340: Documentation on net-core serial recovery

### PR DESCRIPTION
**Pull request by nvlsianpu**
Added elaboration on MCUboot's serial recovery of
the Networking core image.
This covers either "Serial recovery for the NET-core
in the one image pair mode" as "Simultaneous multi-image DFU"
on nRF5340.

This documents feature which is adding by #8050, DNM before it.

### CODE REVIEW

File: doc/nrf/ug_nrf5340.rst
Comment (peknis) [inline doc/nrf/ug_nrf5340.rst:572]:
```suggestion
* :kconfig:option:`CONFIG_NRF53_MULTI_IMAGE_UPDATE` - Select the simultaneous multi-image DFU feature.
```
Suggesting to move this option to the beginning of the list above.

File: doc/nrf/ug_nrf5340.rst
Comment (peknis) [inline doc/nrf/ug_nrf5340.rst:594]:
```suggestion
MCUboot's serial recovery of the networking core image
```

File: doc/nrf/ug_nrf5340.rst
Comment (peknis) [inline doc/nrf/ug_nrf5340.rst:598]:
```suggestion
In addition to the recovery of the application core image, also the networking core image can be recovered.
You can use this feature if you build nRF53 MCUboot with one of the following options:
```

File: doc/nrf/ug_nrf5340.rst
Comment (peknis) [inline doc/nrf/ug_nrf5340.rst:601]:
```suggestion
* :kconfig:option:`CONFIG_NRF53_MULTI_IMAGE_UPDATE` - Simultaneous multi-image DFU.
* :kconfig:option:`CONFIG_NRF53_RECOVERY_NETWORK_CORE` - Serial recovery for the network core in the one image pair mode.
```

File: doc/nrf/ug_nrf5340.rst
Comment (peknis) [inline doc/nrf/ug_nrf5340.rst:603]:
```suggestion
To upload the networking image, use the following command::
```

File: doc/nrf/ug_nrf5340.rst
Comment (peknis) [inline doc/nrf/ug_nrf5340.rst:607]:
```suggestion
``serial_conn`` is the serial connection configuration.
```

File: doc/nrf/ug_nrf5340.rst
Comment (peknis) [inline doc/nrf/ug_nrf5340.rst:609]:
```suggestion
To enable serial recovery of the network core while multi-image update is not enabled in the MCUboot, set the following options:
```

File: doc/nrf/ug_nrf5340.rst
Comment (peknis) [inline doc/nrf/ug_nrf5340.rst:616]:
```suggestion
* select :kconfig:option:`NRF53_RECOVERY_NETWORK_CORE`
```

File: doc/nrf/ug_nrf5340.rst
Comment (peknis) [inline doc/nrf/ug_nrf5340.rst:618]:
```suggestion
Additionally, define and include the memory partitions:
```

File: doc/nrf/ug_nrf5340.rst
Comment (peknis) [inline doc/nrf/ug_nrf5340.rst:622]:
```suggestion
* ``mcuboot_primary`` and ``mcuboot_secondary`` - partitions for the application core image slots.
* ``mcuboot_primary_1`` - partition for the network core image slot.
* ``pcd_sram`` - partition used for command exchange between the application core and the network core (see :kconfig:option:`CONFIG_PCD_APP`).
```

File: doc/nrf/ug_nrf5340.rst
Comment (peknis) [inline doc/nrf/ug_nrf5340.rst:629]:
```suggestion
   When using MCUboot with the :kconfig:option:`CONFIG_NRF53_RECOVERY_NETWORK_CORE` option enabled,
   the application core does not have direct access to the network core flash memory.
   Use ``mcuboot_primary_1`` as the RAM partition mediator.
```

**Review by peknis — CHANGES_REQUESTED** [review]

**Review by pdunaj — APPROVED** [review]
Assuming changes requested by @peknis are applied.

File: doc/nrf/ug_nrf5340.rst
Comment (nvlsianpu) [reply doc/nrf/ug_nrf5340.rst:572]: I should have meant that: After that the feature will be <b>automatically</b> selected by default thanks to ...
So there is no real need to enable CONFIG_NRF53_MULTI_IMAGE_UPDAT manually. I just wan't to mention the option name so it became visible to the user.

File: doc/nrf/ug_nrf5340.rst
Comment (nvlsianpu) [reply doc/nrf/ug_nrf5340.rst:629]: This is explanation of reason because of which mcuboot_primary_1 must exist.
The suggestion changes the meaning.

Comment (peknis) [reply doc/nrf/ug_nrf5340.rst:629]: Sorry for messing up the meaning. How about then something like: "Due to this, ``mcuboot_primary_1`` is used as the RAM partition mediator".

File: doc/nrf/ug_nrf5340.rst
Comment (peknis) [reply doc/nrf/ug_nrf5340.rst:572]: Okay. How about something like this then: "After setting these options, the :kconfig:option:`CONFIG_NRF53_MULTI_IMAGE_UPDATE` option selects this feature automatically by default."

Comment (nvlsianpu) [reply doc/nrf/ug_nrf5340.rst]: applied

File: doc/nrf/ug_nrf5340.rst
Comment (nvlsianpu) [reply doc/nrf/ug_nrf5340.rst]: applied

**Review by peknis — APPROVED** [review]

File: doc/nrf/ug_nrf5340.rst
Comment (greg-fer) [inline doc/nrf/ug_nrf5340.rst:597]: Do we have the application core image recovery described anywhere? This reference to the application core image begs for a link here...

File: doc/nrf/ug_nrf5340.rst
Comment (greg-fer) [inline doc/nrf/ug_nrf5340.rst:598]:
```suggestion
When you build MCUboot for the nRF5340 DK, you can use this feature with one of the following options:
```

File: doc/nrf/ug_nrf5340.rst
Comment (greg-fer) [inline doc/nrf/ug_nrf5340.rst:594]: Two doubts about this section:

- Shouldn't this section be part of the MCUboot documentation? After all, recovering different cores can be a feature not restricted to only nRF5340 DK. In fact, we have also the guide for Thingy:53, so wouldn't this information be valid there as well?
- Shouldn't this section be a subsection of Simultaneous multi-image DFU, not a top section?

File: doc/nrf/ug_nrf5340.rst
Comment (greg-fer) [inline doc/nrf/ug_nrf5340.rst:601]: What is the "one image pair mode" and where can I find more info about it? I see no results when looking for it in MCUboot docs and NCS docs...

File: doc/nrf/ug_nrf5340.rst
Comment (greg-fer) [reply doc/nrf/ug_nrf5340.rst:625]: No.

File: doc/nrf/ug_nrf5340.rst
Comment (greg-fer) [inline doc/nrf/ug_nrf5340.rst:631]: Is this section related to the serial recovery of the network core? It is now set to become a subsection of that chapter.

File: doc/nrf/ug_nrf5340.rst
Comment (greg-fer) [inline doc/nrf/ug_nrf5340.rst:622]:
```suggestion
* ``mcuboot_primary`` and ``mcuboot_secondary`` - Partitions for the application core image slots.
* ``mcuboot_primary_1`` - Partition for the network core image slot.
* ``pcd_sram`` - Partition for command exchange between the application core and the network core (see :kconfig:option:`CONFIG_PCD_APP`).
```

File: doc/nrf/ug_nrf5340.rst
Comment (greg-fer) [inline doc/nrf/ug_nrf5340.rst:609]: Where can I find information about enabling serial recovery in the MCUboot docs?
I see we don't have much about this in the NCS docs either, so maybe we should adapt https://developer.nordicsemi.com/nRF_Connect_SDK/doc/latest/nrf/applications/nrf_desktop/README.html#serial-recovery-dfu for the general docs? @MarekPieta , @pdunaj , opinions?

**Review by greg-fer — CHANGES_REQUESTED** [review]
I have some serious doubts about the missing context of the new section. Going to add @FrancescoSer as reviewer, as he used to take care of the MCUboot docs some time ago, so might know a thing or two...

File: doc/nrf/ug_nrf5340.rst
Comment (FrancescoSer) [reply doc/nrf/ug_nrf5340.rst:594]: I'm guessing this depends on CONFIG_NRF53_RECOVERY_NETWORK_CORE which should be strictly related to the 5340. @nvlsianpu ?

But yeah, I agree this should go in as a subsection

Comment (nvlsianpu) [reply doc/nrf/ug_nrf5340.rst:594]: Section is specific to nRF5340 mcuboot downstream feature. Wher I should place it then?

Shouldn't this section be a subsection of Simultaneous multi-image DFU, not a top section?
No, the simultaneous multi-image DFU is one of two way to make this feature enabled.

**Comment by greg-fer** [comment]
Is this going to 2.1.0, @nvlsianpu ?

File: doc/nrf/ug_nrf5340.rst
Comment (nvlsianpu) [reply doc/nrf/ug_nrf5340.rst:594]: ^^
@FrancescoSer I put the documentation update in this file only because it already contains documentation on CONFIG_NRF53_MULTI_IMAGE_UPDAT.

File: doc/nrf/ug_nrf5340.rst
Comment (nvlsianpu) [reply doc/nrf/ug_nrf5340.rst:597]: Serial recovery is the same serial protocol as used by the mcumgr/smp_svr sample. We don't have any doc on that,

**Comment by nvlsianpu** [comment]
@greg-fer 

> Is this going to 2.1.0, @nvlsianpu ?

Yes as documented feature went in.

File: doc/nrf/ug_nrf5340.rst
Comment (nvlsianpu) [reply doc/nrf/ug_nrf5340.rst:572]: @MarekPieta What about sth liike:
`After setting these options (and all other dependences), the `

File: doc/nrf/ug_nrf5340.rst
Comment (nvlsianpu) [reply doc/nrf/ug_nrf5340.rst:601]: one image pair mode is the mode when UPDATEABLE_IMAGE_NUMBER == 1, so most popular dual-bank mode.

File: doc/nrf/ug_nrf5340.rst
Comment (MarekPieta) [reply doc/nrf/ug_nrf5340.rst:598]: Discussed offline.

File: doc/nrf/ug_nrf5340.rst
Comment (MarekPieta) [reply doc/nrf/ug_nrf5340.rst:609]: Such a generic section would be definitely nice to have (maybe it could be even part of MCUboot's docs).
The docs of serial recovery in nRF Desktop could be used as a base (we would just need to remove references which are specific to nRF Desktop application).

**Comment by greg-fer** [comment]
@nvlsianpu , I'll report the issue about the user guide, but we won't be able to fix this for this release.

When it comes to this PR, @FrancescoSer is working with Vestavind, so please wait for his approval for this PR.

**Comment by nvlsianpu** [comment]
@FrancescoSer Can you review?

File: doc/nrf/ug_nrf5340.rst
Comment (FrancescoSer) [reply doc/nrf/ug_nrf5340.rst:597]: Isn't this the protocol description? https://developer.nordicsemi.com/nRF_Connect_SDK/doc/latest/zephyr/services/device_mgmt/smp_protocol.html

File: doc/nrf/ug_nrf5340.rst
Comment (FrancescoSer) [reply doc/nrf/ug_nrf5340.rst:609]: Maybe a section regarding serial recovery could fit in ug_fw_update.rst, but I see this as a separate task from this PR.

File: doc/nrf/ug_nrf5340.rst
Comment (FrancescoSer) [inline doc/nrf/ug_nrf5340.rst:572]:
```suggestion
The :kconfig:option:`CONFIG_NRF53_MULTI_IMAGE_UPDATE` option selects this feature by default if these options and all its other dependencies are asserted.
```

File: doc/nrf/ug_nrf5340.rst
Comment (FrancescoSer) [reply doc/nrf/ug_nrf5340.rst:598]: Please apply this

File: doc/nrf/ug_nrf5340.rst
Comment (FrancescoSer) [reply doc/nrf/ug_nrf5340.rst:601]:
```suggestion
* :kconfig:option:`CONFIG_NRF53_RECOVERY_NETWORK_CORE` - Serial recovery for the network core in the one image pair mode, where :kconfig:option:`CONFIG_UPDATEABLE_IMAGE_NUMBER` is set to ``== 1``.
  
```

File: doc/nrf/ug_nrf5340.rst
Comment (FrancescoSer) [inline doc/nrf/ug_nrf5340.rst:608]:
```suggestion
``serial_conn`` is the serial connection configuration.
For more information on MCUmgr image management, see :ref:`zephyr:image_mgmt`

```

File: doc/nrf/ug_nrf5340.rst
Comment (FrancescoSer) [inline doc/nrf/ug_nrf5340.rst:609]:
```suggestion
To enable the serial recovery of the network core while the multi-image update is not enabled in the MCUboot, set the following options:
```

**Review by FrancescoSer — CHANGES_REQUESTED** [review]

**Comment by FrancescoSer** [comment]
> @greg-fer @FrancescoSer Looks like ug_nrf5340.rst is for documenting work with both nRF5340dk and thingy53. Shouldn't the tittle be changed?

As this might require deeper edits of ug_5340, I would keep a reshuffling of the content between ug_5340 and ug_thingy53_gs as a separate task. I'll open a JIRA about that.

File: doc/nrf/ug_nrf5340.rst
Comment (nvlsianpu) [reply doc/nrf/ug_nrf5340.rst:598]: @FrancescoSer These options might be available for both nRF5340DK and thingy53 (or any else nRF53 board if user can meet dependencies).

Comment (FrancescoSer) [reply doc/nrf/ug_nrf5340.rst:598]:
```suggestion
When you build MCUboot for the nRF5340 DK or the Thingy:53, you can use this feature with one of the following options:
```

Ok so let's edit this a bit for now, and we will see to move the content in the thingy:53 guide after this PR gets merged

File: doc/nrf/ug_nrf5340.rst
Comment (FrancescoSer) [inline doc/nrf/ug_nrf5340.rst:619]:
```suggestion
Additionally, define and include the following memory partitions:
```

File: doc/nrf/ug_nrf5340.rst
Comment (FrancescoSer) [reply doc/nrf/ug_nrf5340.rst:630]: @nvlsianpu Please check Greg's comment

**Review by FrancescoSer — CHANGES_REQUESTED** [review]

File: doc/nrf/ug_nrf5340.rst
Comment (nvlsianpu) [reply doc/nrf/ug_nrf5340.rst:630]: @FrancescoSer `Container for firmware update binaries` section is related to `Simultaneous multi-image DFU` and DFU in general.

**Review by FrancescoSer — APPROVED** [review]

**Pull request merged**
